.github/workflows/aws-deployment.yml

```diff
@@ -1,38 +1,86 @@
+    name: Build and deploy Todo App
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions/cache@v1
+
+      - name: Checkout code
+        uses: actions/checkout@v1
+
+      - name: Gradle cache
+        uses: actions/cache@v1
         with:
           path: ~/.gradle/caches
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
+
       - name: Set up JDK 11
         uses: actions/setup-java@v1
         with:
           java-version: 11.0
+
       - name: Build application
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+        run: |
+          cd application
+          ./gradlew build --stacktrace
+
+      - name: "Zip build reports"
+        if: failure()
+        run: zip -r reports.zip **/build/reports
+
+      - name: "Upload build reports"
+        uses: actions/upload-artifact@v1
+        if: failure()
+        with:
+          name: reports
+          path: reports.zip
+
+      - name: Create or update ECR registry stack
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: ${{ secrets.AWS_REGION }}
+        run: |
+          cd cloudformation/application-setup-basic/stacks/registry
+          ./create-or-update-registry-stack.sh
+
+      - name: Create or update Cognito stack
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: ${{ secrets.AWS_REGION }}
+        run: |
+          cd cloudformation/application-setup-basic/stacks/cognito
+          ./create-or-update-cognito-stack.sh
+
+      - name: Publish Docker image to ECR registry
         env:
           AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
           AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          AWS_REGION: ${{ secrets.REGION }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
         run: |
-          export TODO_COGNITO_USER_POOL_ID=$(aws cloudformation describe-stacks --region ${AWS_REGION} --stack-name aws101-cognito-user-pool --output text --query 'Stacks[0].Outputs[?OutputKey==`UserPoolId`].OutputValue')
-          export TODO_COGNITO_CLIENT_ID=$(aws cloudformation describe-stacks --region ${AWS_REGION} --stack-name aws101-cognito-user-pool --output text --query 'Stacks[0].Outputs[?OutputKey==`UserPoolClientId`].OutputValue')
-          export TODO_COGNITO_CLIENT_SECRET=$(aws cognito-idp describe-user-pool-client --region ${AWS_REGION} --user-pool-id ${TODO_COGNITO_USER_POOL_ID} --client-id ${TODO_COGNITO_CLIENT_ID} --output text --query 'UserPoolClient.ClientSecret')
```

Comment:
I changed the service stack so that the UserPoolId and the ClientId will be taken directly from the Cognito stack's output values. Doesn't work with the secret, though, sadly, because the CloudFormation resources doesn't expose it as an attribute.